### PR TITLE
Merge `request_decode_func` and `return_value_decode_func`

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
This PR merges `request_decode_func` and `return_value_decode_func` into a single function, since they were actually identical functions. This new function is named: `decode_non_streamed_parameters_func`.

It was fairly small, and after this change, `decode_non_streamed_parameters` was only called in one spot, in this function, so I also went ahead and inlined the function there.

### CODE REVIEW

File: tools/slicec-cs/src/decoding.rs [hunk 1]
```diff
@@ -468,39 +468,60 @@ fn decode_func_body(type_ref: &TypeRef, namespace: &str, encoding: Encoding) ->
...
+            format!(
+                "\
+    (ref SliceDecoder decoder) =>
+    {{
+        {body_content}
+    }}",
+            )
+            .into()
+        }
+    }
```
Comment (externl) [inline tools/slicec-cs/src/decoding.rs:524]: This formatting looks dubious. It should be left indented all the way. You should review the generated code to see that it looks correct.

**Review by externl — APPROVED** [review]
See comment

File: tools/slicec-cs/src/decoding.rs [hunk 2]
```diff
@@ -468,39 +468,60 @@ fn decode_func_body(type_ref: &TypeRef, namespace: &str, encoding: Encoding) ->
...
+            }
+            writeln!(code, "return {};", non_streamed_parameters.to_argument_tuple());
+
+            let body_content = code.indent();
+            format!(
+                "\
+    (ref SliceDecoder decoder) =>
+    {{
+        {body_content}
+    }}",
```
Comment (InsertCreativityHere) [inline tools/slicec-cs/src/decoding.rs:520]:
```suggestion
(ref SliceDecoder decoder) =>
{{
    {body_content}
}}",
```

**Review by InsertCreativityHere — COMMENTED** [review]
Looks good

File: tools/slicec-cs/src/decoding.rs (hunk 1 above)
Comment (InsertCreativityHere) [reply tools/slicec-cs/src/decoding.rs]: Yes, good catch!
It got auto-indented when I copy/pasted this function, and I didn't see it.

**Pull request merged**